net: rename internal functions for readability

joyeecheung opened the pull request:
The current namings are a bit confusing and hard to be searched for.

* Rename listen to listenInCluster
* Rename _listen2 to _setUpListenHandle
* Remove _listen since it's a one-liner only used in one place
* Correct comments in server.listen

Also `Socket.prototype.listen` doesn't seem to be [tested](https://coverage.nodejs.org/coverage-e2f151f5b2e76b94/root/net.js.html) nor documented, added a FIXME.

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
net

joyeecheung commented on lib/net.js:
```diff
@@ -1310,8 +1307,10 @@ Server.prototype._listen2 = function(address, port, addressType, backlog, fd) {
+}
 
+Server.prototype._setUpListenHandle = setUpListenHandle;
+Server.prototype._listen2 = setUpListenHandle;  // legacy alias
```
Not sure if `_listen2` should be simply removed (would anyone rely on a method with a name like this?)...`_setUpListenHandle` can be put into a internal symbol, though.

silverwind replied:
https://github.com/search?l=JavaScript&p=1&q=%22._listen2%22&ref=searchresults&type=Code&utf8=%E2%9C%93 shows a few uses, but nothing major I suppose.

jasnell replied:
Removing it would likely (unfortunately) require a semver-major and a deprecation cycle.

joyeecheung replied:
@jasnell I'll keep it and make a following doc-deprecate PR.

jasnell commented on lib/net.js:
```diff
@@ -1252,19 +1246,19 @@ function createServerHandle(address, port, addressType, fd) {
-
-Server.prototype._listen2 = function(address, port, addressType, backlog, fd) {
-  debug('listen2', address, port, addressType, backlog, fd);
+function setUpListenHandle(address, port, addressType, backlog, fd) {
```
It may just be me but `setUp` looks odd. Perhaps `setupListenHandle` would be better?

jasnell commented on lib/net.js:
```diff
@@ -1325,25 +1324,29 @@ function emitListeningNT(self) {
 
-  if (cluster.isMaster || exclusive) {
-    self._listen2(address, port, addressType, backlog, fd);
+  if (cluster.isMaster || exclusive) { // Will create a new handle
```
Can this comment be moved to the next line on it's own?

jasnell commented on lib/net.js:
```diff
@@ -1433,20 +1438,22 @@ Server.prototype.listen = function() {
 
 function lookupAndListen(self, port, address, backlog, exclusive) {
-  require('dns').lookup(address, function doListening(err, ip, addressType) {
+  require('dns').lookup(address, function doListen(err, ip, addressType) {
```
I know it was already like this, but can you split this into two lines?
```
const dns = require('dns');
dns.lookup(...)
```

joyeecheung commented:
Aside: https://github.com/newrelic/node-newrelic/blob/master/lib/instrumentation/core/net.js is wrapping `_listen2`

joyeecheung commented:
Addressed comments from @jasnell . I decided to leave `_setUpListenHandle` exposed because this might be useful for APM agents. cc @nodejs/diagnostics 

New CI: https://ci.nodejs.org/job/node-test-pull-request/6995/

jasnell commented:
Given that `_listen2` is still exposed as an alias for `_setUpListenHandle`, there is really no reason to also expose `_setUpListenHandle` -- in fact it's likely something we *shouldn't* do given that it's intended to be an internal function.

joyeecheung commented:
@jasnell Yeah...thinking again if `_listen2` has to be exposed for wrapping purposes, then it can be left as-is.

joyeecheung commented:
I just realized if we stopped calling `_listen2` in our code, the agents will be broken anyway, because they wrap `_listen2` under the expectation that this method will be called by Node and they will have a chance to execute their own code in the wrapper. Therefore I only renamed the function declaration and still call `server._listen2` to make the behavioral change minimal in this PR.

New CI: https://ci.nodejs.org/job/node-test-pull-request/7015/

watson commented:
@joyeecheung thanks for thinking of the APM vendors 😄

As far as I can see you ended up aliasing `_setUpListenHandle` as `_listen2`, but still call `_listen2` internally - so existing code expecting a `_listen2` function to be called should still work for now 👍

Both New Relic and Opbeat (and most likely other vendors as well) base our patching of Node core on the [async-listener](https://github.com/othiym23/async-listener) module. I'm not sure how many APM vendors merge in new changes from this module though (as we usually have to modify the module quite a bit to optimize the performance to our use-case). But in any case I've opened a PR in async-listener (https://github.com/othiym23/async-listener/pull/103) to make sure it patches `_setUpListenHandle` instead of of `_listen2` if available.

I fear that the new code is a little harder to understand than the old, since the code is now calling a legacy alias, which to outsiders might seem pretty weird.

So the reason why I want to get this into the hands of APM vendors ASAP is because I fear that it's just a matter of time before someone cleans up the code or by accident just calls `_setUpListenHandle` directly. This will break our instrumentation of course.

joyeecheung commented:
git am fails due to conflicts...rebased and last CI before landing: https://ci.nodejs.org/job/node-test-pull-request/7151/

joyeecheung commented:
Landed in ea4570, thanks!

joyeecheung commented:
EDIT: posted the wrong commit hash, it's 0ea4570

MylesBorins commented:
this should likely be backported

MylesBorins commented:
ping

joyeecheung commented:
Depends on https://github.com/nodejs/node/pull/11667 which is don't-land-on-v6.x
